src/token/components/TokenSelectDropdown.tsx

```diff
@@ -12,58 +12,28 @@ export function TokenSelectDropdown({
   setToken,
   token,
 }: TokenSelectDropdownReact) {
-  const componentTheme = useTheme();
-
   const [isOpen, setIsOpen] = useState(false);
-
-  const handleToggle = useCallback(() => {
-    setIsOpen(!isOpen);
-  }, [isOpen]);
-
-  const dropdownRef = useRef<HTMLDivElement>(null);
   const buttonRef = useRef<HTMLButtonElement>(null);
 
-  /* v8 ignore next 11 */
-  const handleBlur = useCallback((event: MouseEvent) => {
-    const isOutsideDropdown =
-      dropdownRef.current &&
-      !dropdownRef.current.contains(event.target as Node);
-    const isOutsideButton =
-      buttonRef.current && !buttonRef.current.contains(event.target as Node);
-
-    if (isOutsideDropdown && isOutsideButton) {
-      setIsOpen(false);
-    }
-  }, []);
-
-  useEffect(() => {
-    // NOTE: this ensures that handleBlur doesn't get called on initial mount
-    //       We need to use non-div elements to properly handle onblur events
-    setTimeout(() => {
-      document.addEventListener('click', handleBlur);
-    }, 0);
-
-    return () => {
-      document.removeEventListener('click', handleBlur);
-    };
-  }, [handleBlur]);
-
   return (
     <div className="relative max-w-fit shrink-0">
       <TokenSelectButton
         ref={buttonRef}
-        onClick={handleToggle}
+        onClick={() => setIsOpen(!isOpen)}
```

Comment:
better to use `setIsOpen(prev => !prev)`. also probably better to use a callback